pkg/build/util/generate.go

```diff
@@ -45,6 +49,69 @@ func GenerateBuildFromBuild(build *buildapi.Build) *buildapi.Build {
 	}
 }
 
+// GenerateBuildWithImageTag generates a build definition based on the current imageid
+// from any ImageRepository that is associated to the BuildConfig by an ImageChangeTrigger.
+// Takes a BuildConfig to base the build on, an optional SourceRevision to build, and an optional
+// Client to use to get ImageRepositories to check for affiliation to this BuildConfig (by way of
+// an ImageChangeTrigger).  If there is a match in the image repo list, the resulting build will use
+// the image tag from the corresponding image repo rather than the image field from the buildconfig
+// as the base image for the build.
+func GenerateBuildWithImageTag(config *buildapi.BuildConfig, revision *buildapi.SourceRevision, imageRepoGetter osclient.ImageRepositoryNamespaceGetter) (*buildapi.Build, error) {
+
+	imageSubstitutions := make(map[string]string)
+	glog.V(4).Infof("Generating tagged build for config %s", config.Name)
+
+	for _, trigger := range config.Triggers {
+		if trigger.Type != buildapi.ImageChangeBuildTriggerType {
+			continue
+		}
+		icTrigger := trigger.ImageChange
+		glog.V(4).Infof("Found image change trigger with reference to repo %s", icTrigger.From.Name)
+
+		var imageRepo *imageapi.ImageRepository
+		if len(icTrigger.From.Namespace) != 0 {
+			var err error
+			imageRepo, err = imageRepoGetter.GetByNamespace(icTrigger.From.Namespace, icTrigger.From.Name)
+			if err != nil {
+				if errors.IsNotFound(err) {
+					continue
+				}
+				return nil, err
+			}
+		} else {
+			var err error
```

Comment:
why not to move the 'var err' on top of 'if' since you are consuming it in both cases?